packages/server/lib/controllers/connection/getConnections.ts

```diff
@@ -1,20 +1,30 @@
 import { asyncWrapper } from '../../utils/asyncWrapper.js';
-import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { zodErrorToHTTP } from '@nangohq/utils';
 import type { GetPublicConnections } from '@nangohq/types';
 import { AnalyticsTypes, analytics, connectionService } from '@nangohq/shared';
 import { connectionToPublicApi } from '../../formatters/connection.js';
+import { z } from 'zod';
+
+const validationQuery = z
+    .object({
+        connectionId: z.string().max(255).optional()
+    })
+    .strict();
 
 export const getPublicConnections = asyncWrapper<GetPublicConnections>(async (req, res) => {
-    const emptyQuery = requireEmptyQuery(req);
-    if (emptyQuery) {
-        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+    const queryParamValues = validationQuery.safeParse(req.query);
+    if (!queryParamValues.success) {
+        res.status(400).send({
+            error: { code: 'invalid_query_params', errors: zodErrorToHTTP(queryParamValues.error) }
+        });
         return;
     }
 
     const { environment, account } = res.locals;
+    const queryParam: GetPublicConnections['Querystring'] = queryParamValues.data;
 
     void analytics.track(AnalyticsTypes.CONNECTION_LIST_FETCHED, account.id);
-    const connections = await connectionService.listConnections({ environmentId: environment.id, limit: 10000 });
+    const connections = await connectionService.listConnections({ environmentId: environment.id, search: queryParam.connectionId, limit: 10000 });
```

Comment:
as you mention in the description, it might be more flexible and understandable to have `search` be of type `{connection?: {id: string}, end_user?: {name?: string, email?: string}` and only search if attributes is set but that's way more complicated

Reply:
yeah and at the same time search is usually a wildcard, not sure I can fine tune after when Bastien answers